openhands/runtime/plugins/agent_skills/file_editor/impl.py

```diff
@@ -211,7 +211,9 @@ def insert(self, path: Path, insert_line: int, new_str: str):
         snippet_lines = (
             file_text_lines[max(0, insert_line - SNIPPET_LINES) : insert_line]
             + new_str_lines
-            + file_text_lines[insert_line : insert_line + SNIPPET_LINES]
+            + file_text_lines[
+                insert_line : min(n_lines_file, insert_line + SNIPPET_LINES)
+            ]
```

Comment:
Nit: I think this might not be necessary -- the second index can actually exceed the total length of the list i believe

Reply:
Ah yeah, just tried it, Python does forgive with out-of-range slice indices